src/utils/augmentation.py

```diff
@@ -0,0 +1,132 @@
+import random
+import numpy as np
+
+from abc import abstractmethod, ABC
+
+
+class AugmentationWord(ABC):
+    @abstractmethod
+    def __call__(self, *args, **kwargs):
```

Comment:
`__call__` always accept single word and return single word. So you can define here
```python
def __call__(self, word: str) -> str
```